packages/TagPicker/src/TagPicker.tsx

```diff
@@ -118,12 +118,23 @@ const TagPicker: React.FunctionComponent<TagPickerProps> = ({
     const [keyboardFocusIndex, setKeyboardFocusIndex] = useState(
         defaultKeyboardFocusIndex
     );
+    const selectedResults = maxTags ? selectedResultsProp.slice(0, maxTags) : selectedResultsProp;
     const selectedResultsCount = selectedResults.length;
+    const hasTooManyTags = selectedResults.length !== selectedResultsProp.length;
+
+    hasTooManyTags && console.warn(
```

Comment:
Is it necessary to display a warning in the console for this use case, as the information is already being returned to the user?